### PR TITLE
Azure: Deploy Elastic Agent

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes
This adds a script to deploy the Elastic Agent. When the script fails, Azure displays stdout and stderr directly on the deployment page. The script is hosted directly in our github repo.

### Screenshot/Data
image

### Related Issues
- Related: https://github.com/elastic/cloudbeat/issues/1248

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -75,6 +75,8 @@ repos:
     hooks:
       - id: shfmt
         args: ["-i", "4"]
+      - id: shellcheck
+        files: deploy/azure/.*sh
```
Comment (orestisfl) [inline .pre-commit-config.yaml:79]: We can add this to other scripts but most of them fail so I've added it just for this script as an opportunity to use it in new scripts. We can track improvements on the rest on a new ticket.

This is IMO a good idea, especially for public-facing scripts that are run by users.

File: deploy/azure/install-agent.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -euxo pipefail
+
+usage="$0 <elastic agent version> <elastic artifact server> <fleet url> <enrollment token>"
+ElasticAgentVersion=${1:?$usage}
+ElasticArtifactServer=${2:?$usage}
+FleetUrl=${3:?$usage}
+EnrollmentToken=${4:?$usage}
+
+ElasticAgentArtifact="elastic-agent-$ElasticAgentVersion-linux-x86_64"
```
Comment (orestisfl) [inline deploy/azure/install-agent.sh:10]: I went with x86 here, should we use arm for azure as well for consistency?

File: deploy/azure/install-agent.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+curl -L -O "${ElasticArtifactServer}/$ElasticAgentArtifact.tar.gz"
+tar xzf "${ElasticAgentArtifact}.tar.gz"
+cd "${ElasticAgentArtifact}"
+sudo ./elastic-agent install --non-interactive --url="${FleetUrl}" --enrollment-token="${EnrollmentToken}"
```
Comment (orestisfl) [inline deploy/azure/install-agent.sh:14]: This is missing the `--tag` arguments but I am not sure if we have anything to add here yet.

File: deploy/azure/install-agent.sh (hunk 2 above)
Comment (oren-zohar) [reply deploy/azure/install-agent.sh:10]: Most of our test if not all of them are on Linux x86 machines AFAIK, so I think it's better to have it like that

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**